src/rf/js/src/home/components/modals.js

```diff
@@ -122,30 +124,50 @@ var UploadModal = React.createBackboneClass({
         return fileNameParts[fileNameParts.length - 1];
     },
 
-    s3BucketKeyValid: function(s3BucketKey) {
-        // example valid s3BucketKey: ab-cd.ef/gh1/i-j.k
-        var re = /^([\w-.]+\/)+[\w-]+\.\w+$/;
-        return re.test(s3BucketKey);
+    isTiff: function(fileName) {
+        var extension = this.getFileExtension(fileName).toLowerCase();
+        return extension === 'tif' || extension === 'tiff';
+    },
+
+    s3URIValid: function(s3URI) {
+        // example valid s3URI
+        // s3://bucket-name/path/to/tiff.tif
+        var re = /^s3\:\/\/([\w-.]+\/)+[\w-]+\.\w+$/;
```

Comment:
The normal http format for s3 buckets seems equally natural. If I have an S3 bucket, that's what it shows me. I think we should consider supporting this as well.

![screenshot from 2015-10-26 17 07 37](https://cloud.githubusercontent.com/assets/903219/10742545/17c3d04c-7c04-11e5-9d08-357eb460fccd.png)